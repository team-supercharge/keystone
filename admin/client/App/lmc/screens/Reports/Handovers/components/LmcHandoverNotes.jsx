import React, { Component } from 'react'
import PropTypes from 'prop-types'
import LmcHandoverNoteCard from './LmcHandoverNoteCard.jsx';

export default class LmcHandoverNotes extends Component {
    render() {
        return (
            <div style={styles.mainContainer}>
                { this.props.notes.map((note, i) => {
                    return (
                        <LmcHandoverNoteCard
                            key={i}
                            note={note}
                        />
                    )
                })}
                {// if we want to edit notes (i.e. on current handover), 
                // we can render this here if props.edit === true
                }
            </div>
    
            
        )
    }
}

const styles = {
    mainContainer: {
        display: 'flex',
        flexDirection: 'column'
    },
}

LmcHandoverNotes.propTypes = {
    notes: PropTypes.array.isRequired
}