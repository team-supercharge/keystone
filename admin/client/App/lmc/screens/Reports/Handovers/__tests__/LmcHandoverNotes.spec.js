import React from 'react'
import { shallow } from 'enzyme'
import LmcHandoverNotes from '../components/LmcHandoverNotes.jsx'

describe('LmcHandoverNotes', () => {
    let wrapper
    let notes

    beforeEach(() => {
        notes = [
            { id: 'id', createdBy: 'TestCarer', createdOn: new Date('1/1/2019') },
            { id: 'id2', createdBy: 'TestCarer2', createdOn: new Date('1/1/2019') }
        ]
        wrapper = shallow(
            <LmcHandoverNotes
                notes={notes}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders an LmcHandoverNoteCard for each note', () => {
        const firstNote = wrapper.find('LmcHandoverNoteCard').first()
        expect(wrapper.find('LmcHandoverNoteCard').length).toEqual(notes.length)
        expect(firstNote.props().note).toEqual(notes[0])
    })
})