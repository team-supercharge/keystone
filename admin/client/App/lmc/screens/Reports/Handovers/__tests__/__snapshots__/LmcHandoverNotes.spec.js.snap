// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcHandoverNotes renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <LmcHandoverNoteCard
      note={
        Object {
          "createdBy": "TestCarer",
          "createdOn": 2019-01-01T00:00:00.000Z,
          "id": "id",
        }
      }
    />
    <LmcHandoverNoteCard
      note={
        Object {
          "createdBy": "TestCarer2",
          "createdOn": 2019-01-01T00:00:00.000Z,
          "id": "id2",
        }
      }
    />
  </div>,
  "nodes": Array [
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <LmcHandoverNoteCard
        note={
          Object {
            "createdBy": "TestCarer",
            "createdOn": 2019-01-01T00:00:00.000Z,
            "id": "id",
          }
        }
      />
      <LmcHandoverNoteCard
        note={
          Object {
            "createdBy": "TestCarer2",
            "createdOn": 2019-01-01T00:00:00.000Z,
            "id": "id2",
          }
        }
      />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcHandoverNotes
        notes={
          Array [
            Object {
              "createdBy": "TestCarer",
              "createdOn": 2019-01-01T00:00:00.000Z,
              "id": "id",
            },
            Object {
              "createdBy": "TestCarer2",
              "createdOn": 2019-01-01T00:00:00.000Z,
              "id": "id2",
            },
          ]
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcHandoverNotes {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "notes": Array [
            Object {
              "createdBy": "TestCarer",
              "createdOn": 2019-01-01T00:00:00.000Z,
              "id": "id",
            },
            Object {
              "createdBy": "TestCarer2",
              "createdOn": 2019-01-01T00:00:00.000Z,
              "id": "id2",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <LmcHandoverNoteCard
            note={
              Object {
                "createdBy": "TestCarer",
                "createdOn": 2019-01-01T00:00:00.000Z,
                "id": "id",
              }
            }
          />
          <LmcHandoverNoteCard
            note={
              Object {
                "createdBy": "TestCarer2",
                "createdOn": 2019-01-01T00:00:00.000Z,
                "id": "id2",
              }
            }
          />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <LmcHandoverNoteCard
            note={
              Object {
                "createdBy": "TestCarer",
                "createdOn": 2019-01-01T00:00:00.000Z,
                "id": "id",
              }
            }
          />
          <LmcHandoverNoteCard
            note={
              Object {
                "createdBy": "TestCarer2",
                "createdOn": 2019-01-01T00:00:00.000Z,
                "id": "id2",
              }
            }
          />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcHandoverNotes
    notes={
      Array [
        Object {
          "createdBy": "TestCarer",
          "createdOn": 2019-01-01T00:00:00.000Z,
          "id": "id",
        },
        Object {
          "createdBy": "TestCarer2",
          "createdOn": 2019-01-01T00:00:00.000Z,
          "id": "id2",
        },
      ]
    }
  />,
}
`;
